src/texttospeechbar.cpp

```diff
@@ -140,6 +141,8 @@ void TextToSpeechBar::voiceSelected(int index)
     KiwixApp::instance()->saveVoiceName(currentLang, voice.name());
 
     m_speech.setVoice(voice);
+    if (m_speech.state() == QTextToSpeech::Speaking)
+        speak(m_text);
```

Comment:
IMHO, that won't be good UX. If possible (given the limitation of pause/resuming functionality of `QTextToSpeech`) it would be better to pause the speech, change the voice and resume the speech.

Reply:
My first thought as well, unfortunately, QTextToSpeech's pause and resume barely work.